### DISCUSSION TITLE
Support for conversion of raw files that contain multiple frames (Like canon raw movie (CRM) and CR3 dual pixel raw files)

### DISCUSSION THREAD

**Discussion by Leevas** [dnglab/dnglab#218] [Ideas]
Hi, thanks for this brilliant application.
Would it be possible to add support for raw files with multiple frames ? (for example CR3 with dual raw activated(2 frames in one file) and CRM files (Canon raw movie format).
I've been able to convert CRM files (from the Canon R5) to dng, but only the first frame is opened/accessible in raw editors, normally in rawtherapee you can select sub-frames in the raw tab when the file contains multiple frames.
When converting a dual pixel raw CR3 with adobe dng converter, you can select sub frames in rawtherapee on the raw tab. So it seems that the dng format is capable of holding multiple frames.
DNGlab does seem to convert all the raw data from a CRM file, the resulting dng is about as big as the CRM file, but only the first frame is accesible.

**Comment by cytrinox** [comment]
This is possible with *--image-index*:

```
 --image-index <index>
            Select a specific image index (or 'all') if file is a image container [default: 0]
```
If you specifiy **all**, for each CRM frame a separate DNG file is generated. 

For dual pixel CR3 files, there is a workaround to choose another image trak id. You need to set the environment variable **RAWLER_CRX_RAW_TRAK** to the ISOBMFF trak index pointing to the dual pixel image. IIRC it's index 4. If env var is not specified, dnglab automatically selects the image trak id containing the high resolution image (other traks are low-res-raw, jpeg etc.)
